pkg/checker/checker_test.go

```diff
@@ -172,92 +258,127 @@ func TestContainer(t *testing.T) {
 			imageURL:  "localhost:5000/joshvanl/version-checker@sha:123",
 			opts:      new(api.Options),
 			searchResp: &api.ImageTag{
-				Tag: "",
-				SHA: "sha:456",
+				Tag:          "",
+				SHA:          "sha:456",
+				OS:           utils.OsLinux,
+				Architecture: utils.ArchAMD64,
 			},
 			expResult: &Result{
 				CurrentVersion: "sha:123",
 				LatestVersion:  "sha:456",
 				ImageURL:       "localhost:5000/joshvanl/version-checker",
 				IsLatest:       false,
+				OS:             utils.OsLinux,
+				Architecture:   utils.ArchAMD64,
 			},
+			nodes: []*corev1.Node{utils.CreateNode("node1", utils.ArchAMD64, utils.OsLinux)},
+			pods:  []*corev1.Pod{utils.CreatePodWithNode("pod1", "node1")},
 		},
 		"if using sha and is latest, return true": {
 			statusSHA: "localhost:5000/version-checker@sha:123",
 			imageURL:  "localhost:5000/joshvanl/version-checker@sha:123",
 			opts:      new(api.Options),
 			searchResp: &api.ImageTag{
-				Tag: "v0.2.0",
-				SHA: "sha:123",
+				Tag:          "v0.2.0",
+				SHA:          "sha:123",
+				OS:           utils.OsLinux,
+				Architecture: utils.ArchAMD64,
 			},
 			expResult: &Result{
 				CurrentVersion: "sha:123",
 				LatestVersion:  "v0.2.0@sha:123",
 				ImageURL:       "localhost:5000/joshvanl/version-checker",
 				IsLatest:       true,
+				OS:             utils.OsLinux,
+				Architecture:   utils.ArchAMD64,
 			},
+			nodes: []*corev1.Node{utils.CreateNode("node1", utils.ArchAMD64, utils.OsLinux)},
+			pods:  []*corev1.Pod{utils.CreatePodWithNode("pod1", "node1")},
 		},
 		"if using sha and is latest, return true and no tag if non exists": {
 			statusSHA: "localhost:5000/version-checker@sha:123",
 			imageURL:  "localhost:5000/joshvanl/version-checker@sha:123",
 			opts:      new(api.Options),
 			searchResp: &api.ImageTag{
-				Tag: "",
-				SHA: "sha:123",
+				Tag:          "",
+				SHA:          "sha:123",
+				OS:           utils.OsLinux,
+				Architecture: utils.ArchAMD64,
 			},
 			expResult: &Result{
 				CurrentVersion: "sha:123",
 				LatestVersion:  "sha:123",
 				ImageURL:       "localhost:5000/joshvanl/version-checker",
 				IsLatest:       true,
+				OS:             utils.OsLinux,
+				Architecture:   utils.ArchAMD64,
 			},
+			nodes: []*corev1.Node{utils.CreateNode("node1", utils.ArchAMD64, utils.OsLinux)},
+			pods:  []*corev1.Pod{utils.CreatePodWithNode("pod1", "node1")},
 		},
 	}
 
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
+			nodes := architecture.New()
+			for _, node := range test.nodes {
+				nodes.AddNode(node)
+			}
+			checker := New(search.New().With(test.searchResp, nil), nodes)
 
-			checker := New(search.New().With(test.searchResp, nil))
-			pod := &corev1.Pod{
-				Status: corev1.PodStatus{
-					ContainerStatuses: []corev1.ContainerStatus{
-						{
-							Name:    "test-name",
-							ImageID: test.statusSHA,
-						},
+			// take first pod and update the status
+			pod := test.pods[0]
```

Comment:
Since we only ever use the first pod in the list, make test.pods -> test.pod as above